### PR TITLE
Run tsc before mocha in `npm run fulltest`

### PR DESCRIPTION
**Pull request by TheJetOU**
It's kind of annoying to run 5 minutes of mocha test just for the test to fail in the TypeScript compiler which is at least 5x times faster.

### CODE REVIEW

**Comment by scheibo** [comment]
Use `mocha run test` instead of `mocha run fulltest` during iterative development and then its like 10s tests vs. 20s `tsc` (on my box). And if you want to run `tsc`, then run `tsc`, not `mocha run fulltest`.

**Comment by Zarel** [comment]
The difference in timing is so weird. I moved Mocha first because Mocha was taking ~10s and TypeScript was taking ~20s. But I guess a Mocha fulltest takes longer.

**Comment by Zarel** [comment]
What are you doing using `npm run fulltest` anyway? You should be doing `npm test`. `npm run fulltest` is really only to be done once.

**Comment by TheJetOU** [comment]
My theory was: run `npm run fulltest` -> mocha passes -> `tsc` throws -> fix it -> re-run to make sure everything passes. I could shave off 5 minutes by making `tsc` throw before mocha passes.

**Comment by TheJetOU** [comment]
Oh wow, I wasn't aware of what `npm run test`, where it seems to take sub 90 seconds for the entire test suite. I was running 6 minute tests like a manic the whole fucking time.

I guess, yeah, there's no reason to use `npm run fulltest` most of the time, but I think there's only upsides to this change no matter how minimal.

**Comment by Zarel** [comment]
I'm saying:

1. You should be using `npm test`, not `npm run fulltest`. The full test is just a sanity check; having Travis run it is plenty enough safety for my tastes.

2. You should be using VS Code with live-highlighted `tsc` errors.

**Comment by TheJetOU** [comment]
> You should be using npm test, not npm run fulltest. The full test is just a sanity check; having Travis run it is plenty enough safety for my tastes.

(I'm assuming you haven't seen my previous comment.)

> You should be using VS Code with live-highlighted tsc errors.

VSCode doesn't report errors for _all_ files, only the ones you have currently open, or have opened recently.

**Comment by Zarel** [comment]
Oh, yeah, also it's just `npm test`, the `run` part is optional.

**Comment by Zarel** [comment]
As I said: The reason we currently do `mocha` before `tsc` is because on my laptop (and pre's, apparently), `mocha` takes 10 seconds and `tsc` takes 20 seconds.

File: package.json [hunk 1]
```diff
@@ -25,8 +25,7 @@
     "lint": "eslint --cache . && tslint --project .",
     "pretest": "npm run lint && npm run build",
     "test": "mocha",
```
Comment (Zarel) [inline package.json]: Wait, wait, wait, `test` no longer runs `tsc`????

**Comment by TheJetOU** [comment]
> As I said: The reason we currently do `mocha` before `tsc` is because on my laptop (and pre's, apparently), `mocha` takes 10 seconds and `tsc` takes 20 seconds.

Not for `fulltest` though..

File: package.json (hunk 1 above)
Comment (whalemer) [reply package.json]: `npm test` runs the pretest script too

Comment (Zarel) [reply package.json]: Okay, it does, `npm test` is different from `npm run test`.

Comment (TheJetOU) [reply package.json]: You are correct, it does run the `pretest` script.

**Comment by Zarel** [comment]
Your change affects `test`, though.

**Comment by Zarel** [comment]
I suppose I have no opinion on this in its new state.

@scheibo?

**Pull request merged**